src/mono/wasm/test-main.js

```diff
@@ -342,6 +346,17 @@ if (typeof globalThis.crypto === 'undefined') {
 }
 
 let toAbsoluteUrl = function (path, prefix) {
+    if (is_shell) {
```

Comment:
Also, would it be possible to take the logic and move it from the `*main.js ` inside `dotnet.js` ?
Also for CJS.

Reply:
https://github.com/dotnet/runtime/pull/69441#discussion_r907719667
If the scriptDirectory issue indicated in this comment were resolved, it could be run in a directory other than AppBundle (I think it would still work if it were in AppBundle).

With V8 in the CJS environment, it is not possible to identify the directory where the script is located ( it might be possible, but I don't know how to do it), so porting it to CJS would be difficult.
I assume that I would need to explicitly pass the scriptDirectory to the createDotnetRuntime function, or pass a customized locateFile.

Reply:
I think it's OK that V8 must be run in the current directory in CJS mode.

Reply:
I removed `fs` and `process` module from app-support.mjs and removed the code that `throw Error` for anything other than `is_node` and it works in v8.